### PR TITLE
Crawl Object Refactor

### PR DESCRIPTION
**Pull request by brittainhard**

### CODE REVIEW

**Comment by brittainhard** [comment]
@kriehl @ahmadia this is ready for review.

**Comment by brittainhard** [comment]
Make sure to blow away your current database and run ./app_setup.sh to get a new one.

**Comment by brittainhard** [comment]
Deleteting seeds is currently buggy. Fixing it right now.

File: source/apps/crawl_space/views.py [hunk 1]
```diff
@@ -335,7 +335,7 @@ class DeleteCrawlView(SuccessMessageMixin, ProjectObjectMixin, DeleteView):
 
     def delete(self, request, *args, **kwargs):
         """ Remove crawl folder """
-        shutil.rmtree(self.get_object().get_crawl_path())
+        shutil.rmtree(self.get_object().crawl_location)
```
Comment (ahmadia) [inline source/apps/crawl_space/views.py]: Can we get a little more protection around this function call?  `shutil.rmtree` is equivalent to `rm -rf` and I'd like some validation that it's running this function in the appropriate subdirectory.

Comment (brittainhard) [inline source/apps/crawl_space/views.py]: What sort of validation?

**Pull request merged**